File: javasCE2/ClasesVO/PlaylistVO.java
package javasCE2.ClasesVO;
import java.io.Serializable;
import java.util.Objects;

public class PlaylistVO implements Serializable {
    private static final long serialVersionUID = 1L;

    private String listaId;
    private String nombreCancion;
    private String nombreCreador;

    public PlaylistVO(){}

    public PlaylistVO(String listaId, String nombreCancion, String nombreCreador) {
        this.listaId = listaId;
        this.nombreCancion = nombreCancion;
        this.nombreCreador = nombreCreador;
    }

    public String getListaId() {
        return listaId;
    }

    public void setListaId(String listaId) {
        this.listaId = listaId;
    }

    public String getNombreCancion() {
        return nombreCancion;
    }

    public void setNombreCancion(String nombreCancion) {
        this.nombreCancion = nombreCancion;
    }

    public String getNombreCreador() {
        return nombreCreador;
    }

    public void setNombreCreador(String nombreCreador) {
        this.nombreCreador = nombreCreador;
    }

    @Override
    public String toString() {
        return "PlaylistVO{" +
                "listaId='" + listaId + '\'' +
                ", nombreCancion='" + nombreCancion + '\'' +
                ", nombreCreador='" + nombreCreador + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PlaylistVO that = (PlaylistVO) o;
        return Objects.equals(listaId, that.listaId) &&
                Objects.equals(nombreCancion, that.nombreCancion) &&
                Objects.equals(nombreCreador, that.nombreCreador);
    }

    @Override
    public int hashCode() {
        return Objects.hash(listaId, nombreCancion, nombreCreador);
    }
    
}

File: phps/PlaylistVO.php
<?php
class PlaylistVO {
    
    private $listaId;
    private $nombreCancion;
    private $nombreCreador;

    // Constructor
    public function __construct($listaId = null, $nombreCancion = null, $nombreCreador = null) {
        $this->listaId = $listaId;
        $this->nombreCancion = $nombreCancion;
        $this->nombreCreador = $nombreCreador;
    }

    // Getters
    public function getListaId() {
        return $this->listaId;
    }

    public function getNombreCancion() {
        return $this->nombreCancion;
    }

    public function getNombreCreador() {
        return $this->nombreCreador;
    }

    // Setters
    public function setListaId($listaId) {
        $this->listaId = $listaId;
    }

    public function setNombreCancion($nombreCancion) {
        $this->nombreCancion = $nombreCancion;
    }

    public function setNombreCreador($nombreCreador) {
        $this->nombreCreador = $nombreCreador;
    }

    // toString
    public function toString() {
        return "PlaylistVO{" .
                "listaId='" . $this->listaId . "'" .
                ", nombreCancion='" . $this->nombreCancion . "'" .
                ", nombreCreador='" . $this->nombreCreador . "'" .
                '}';
    }

    // Método para comprobar igualdad basado en todos los atributos
    public function equals($obj) {
        if ($this === $obj) return true;
        if ($obj === null || get_class($this) !== get_class($obj)) return false;
        
        return $this->listaId === $obj->getListaId() &&
               $this->nombreCancion === $obj->getNombreCancion() &&
               $this->nombreCreador === $obj->getNombreCreador();
    }
}
?>